lib/model/src/missions/aar.rs

```diff
@@ -72,13 +72,20 @@ impl Aar {
                 .collect()
         };
 
-        if contractors.len()
-            != contractors
-                .iter()
-                .collect::<std::collections::HashSet<_>>()
-                .len()
-        {
-            return Err("Duplicate contractors found.".to_string());
+        //collect names of duplicate contractors
+        let mut duplicates = Vec::new();
+        for i in 0..contractors.len() {
+            for j in i + 1..contractors.len() {
+                if contractors[i] == contractors[j] {
+                    duplicates.push(contractors[i].clone());
+                }
+            }
+        }
```

Comment:
The HashSet can still be used to know if there are any duplicates, then compare items to find the duplicates.

Rather than compare every item against every other, we can first sort the list, then duplicates will be next to each other, so we can just check each item against the next.

```suggestion
        let unique = contractors.iter().collect::<HashSet<_>>();
        if unique.len() != contractors.len() {
            contractors.sort();
            let mut duplicates = Vec::new();
            for i in contractors {
                  if contractors[i] == contractors[i + 1] {
                      contractors.push(contractors[i].clone());
                  }
            }
        }
```

Reply:
i agree but the list is small enough that it might be faster to not sort and insert into a haashset

Reply:
resolved